### DISCUSSION TITLE
HPC-GAP: ShareObj(x) -- unverified  behavior

### DISCUSSION THREAD

**Discussion by jaslamMiktex** [gap-system/gap#5507] [Q&A] [answered]
There are a few places documented under "debugging HPC-GAP" that do not conform to the actual behavior.
For example, in one place there is a note on protected objects:
```
gap> lx :=[1,2,3];          
[ 1, 2, 3 ]
gap> ShareObj(lx);
<obj 4406326112 inaccessible in region: 0x106aaa540>
gap> lx[1];
Error, Attempt to read object 4406326112 of type list (plain,cyc) without having read access
not in any function at line 4 of stream
```
Whereas, the actual result I am getting is: (accessibility is fine):
```
lx:=[1,2,3];
[ 1, 2, 3 ]
gap> ShareObj(lx);
<protected object in shared region 0x7fd66e474a80 (id: 140558962099376)>
gap> lx[1];
1
```
Please take a look
Thank you,

**Comment by fingolfin** [comment, marked as answer]
This is because region protection (which used to be implemented by the tool `ward` by @rbehrends by preprocessing the GAP kernel source code) is not working anymore and hence is not used anymore.

That means many of the checks that are meant to protect against races etc. are not in place -- so you can "do more" but that "more" unfortunately includes the possibility for corruptions and deadlocks. Be wary. There is a reason HPC-GAP is marked alpha and is not promoted.


(Technically there is PR #2845 which was meant to act as a successor for `ward`, but it was never completed, and it is not clear to me today if it, or something else, ever will be :-/)

**Reply by jaslamMiktex** [reply]
Thank you- it helps, and seems it is a confirmation of the alpha status in the near future.
